getting-started/Frontend/emojify/src/main/java/com/google/cloud/kotlin/emojify/ImageActivity.kt

```diff
@@ -0,0 +1,243 @@
+        queue.add(request)
+    }
+
+    private fun deleteSourceImage() = storageRef.child(imageId).delete()
+            .addOnSuccessListener { Log.i("deleted", "Source image successfully deleted!") }
+            .addOnFailureListener { err -> Log.e("delete", err.message) }
+
+    private fun uploadImage(path: String) {
+        val file = Uri.fromFile(File(path))
+        imageId = System.currentTimeMillis().toString() + ".jpg"
+        val imgRef = storageRef.child(imageId)
+        updateUI {
+            image_view.visibility = View.GONE
+            tv_message!!.text = getString(R.string.waiting_msg_1)
+        }
+        imgRef.putFile(file, StorageMetadata.Builder().setContentType("image/jpg").build())
+                .addOnSuccessListener { _ ->
+                    updateUI { tv_message.text = getString(R.string.waiting_msg_2) }
+                    callEmojifyBackend()
+                }
+                .addOnFailureListener { err ->
+                    updateUI {
+                        show("Cloud Storage error!")
+                        tv_message.text = getString(R.string.storage_error)
+                    }
+                    Log.e("storage", err.message)
+                }
+    }
+
+    private fun downloadAndShowImage() {
+        val url = emjojifiedUrl
+        updateUI {
+            Glide.with(this)
+                    .load(url)
+                    .apply(RequestOptions().signature(ObjectKey(System.currentTimeMillis())))
+                    .apply(RequestOptions()
+                            .placeholder(R.drawable.placeholder)
+                            .error(R.drawable.placeholder)
+                            .diskCacheStrategy(DiskCacheStrategy.NONE)
+                            .dontTransform()
+                            .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
+                            .skipMemoryCache(true))
+                    .into(image_view)
+        }
+        image_view.visibility = View.VISIBLE
+    }
+
+    private fun load(path: String) =
+        launch(CommonPool + job) {
+            uploadImage(path)
+        }
+
+    private fun selectImage() {
+        Album.image(this)
+            .singleChoice()
+            .camera(true)
+            .columnCount(2)
+            .widget(
+                Widget.newDarkBuilder(this)
+                    .title(toolbar!!.title.toString())
+                    .build()
+            )
+            .onResult { result ->
+                albumFiles = result
+                tv_message.visibility = View.VISIBLE
+                if (result.size > 0) load(result[0].path)
+            }
+            .onCancel {
+                finish()
+            }
+            .start()
+    }
+
+    private fun previewImage(position: Int) {
+        if (albumFiles == null || albumFiles!!.size == 0) Toast.makeText(this, R.string.no_selected, Toast.LENGTH_LONG).show()
```

Comment:
Shouldn't the compiler know at this point `albumFiles` isn't null because of the preceding conditional? So `!!` shouldn't be necessary.